lib/Reader/reader.cpp

```diff
@@ -8625,3 +8662,180 @@ void ReaderBase::resolveToken(mdToken Token, CorInfoTokenKind TokenType,
   return resolveToken(Token, getCurrentContext(), getCurrentModuleHandle(),
                       TokenType, ResolvedToken);
 }
+
+#pragma region SIMD_INTRINSICS
+
+//===----------------------------------------------------------------------===//
+//
+// SIMD Intrinsics
+//
+//===----------------------------------------------------------------------===//
+
+IRNode *ReaderBase::generateSIMDBinOp(int OperationCode) {
+  IRNode *Arg2 = ReaderOperandStack->pop();
+  IRNode *Arg1 = ReaderOperandStack->pop();
+  if (checkVectorType(Arg1) && checkVectorType(Arg2)) {
+
+
+    IRNode *Vector1 = Arg1;
+    IRNode *Vector2 = Arg2;
+    
+    IRNode *opResult = 0;
+    switch (OperationCode) {
```

Comment:
Is there a way to map the OperationCode into an enum so we don't have raw integers in the switch?